### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by marshalek37**
[DEMO LINK](https://marshalek37.github.io/react_todo-app-with-api/)

I need help, don'tt know why the loader is not working :(

### CODE REVIEW

**Comment by kasiakubisiak** [comment]
you should be able to cancel edition on escape and keyup, now it doesnt work :/

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,202 @@
...
+          <input
+            type="text"
+            className="todoapp__new-todo"
+            placeholder="What needs to be done?"
+            value={newTodo}
+            onChange={(event) => setNewTodo(event.target.value)}
+            disabled={isLoading}
+          />
+        </form>
+        {tempTodo !== null && <span className="hide">{tempTodo.title}</span>}
```
Comment (kasiakubisiak) [inline src/App.tsx:171]:
```suggestion
        {tempTodo && <span className="hide">{tempTodo.title}</span>}
```

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+const Footer: React.FC<FooterProps>
+= ({
+  todos, filterType, onFilterChange, onClearCompleted,
+}) => {
+  const filteredTodos = useMemo(() => {
+    switch (filterType) {
+      case 'active':
+        return todos.filter((todo) => !todo.completed);
```
Comment (kasiakubisiak) [inline src/components/Footer.tsx:19]: you can create separate function here not to rewrite logic twice

**Review by kasiakubisiak — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,202 @@
...
+import TodoList from './components/TodoList';
+import Header from './components/Header';
 
-const USER_ID = 0;
+export const userId = 10590;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
```
Comment (d-wasilewski) [inline src/App.tsx:13]: Just use empty string if there is no error
```suggestion
  const [error, setError] = useState('');
```

**Review by d-wasilewski — CHANGES_REQUESTED** [review]
This shouldn't completetly disappear, the active todos are still there 
https://github.com/mate-academy/react_todo-app-with-api/assets/71132144/debd827f-8b83-4ab9-9f08-4d9f97a966b9

**Pull request closed without merging**